condition/condition_plugins/CMakeLists.txt

```diff
@@ -1,66 +1,26 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(condition_plugins)
 
-add_compile_options(-std=c++14)
-
-find_package(catkin REQUIRED COMPONENTS
-  pluginlib
-  scenario_conditions
-  scenario_utility
-  scenario_logger
-  scenario_logger_msgs
-)
-
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS
-    pluginlib
-    scenario_conditions
-    scenario_logger
-    scenario_logger_msgs
-    scenario_utility
-    )
-
-include_directories(
-  include
-  ${CMAKE_CURRENT_SOURCE_DIR}
-  ${YAML_CPP_INCLUDE_DIR}
-  ${catkin_INCLUDE_DIRS}
-  )
-
-add_library(${PROJECT_NAME} SHARED
-  # src/accelaration_condition.cpp
-  # src/always_false_condition.cpp
-  # src/always_true_condition.cpp
-  # src/collision_by_entity_condition.cpp
-  # src/reach_position_condition.cpp
-  # src/relative_distance_condition.cpp
-  # src/signal_condition.cpp
-  # src/simulation_time_condition.cpp
-  # src/speed_condition.cpp
-  src/targets.cpp
+if(NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 14)
+endif()
+
+find_package(ament_cmake_auto REQUIRED)
+ament_auto_find_build_dependencies()
+
+ament_auto_add_library(${PROJECT_NAME} SHARED
+   src/accelaration_condition.cpp
+   src/always_false_condition.cpp
+   src/always_true_condition.cpp
+   src/collision_by_entity_condition.cpp
+   src/reach_position_condition.cpp
+   src/relative_distance_condition.cpp
+   src/signal_condition.cpp
+   src/simulation_time_condition.cpp
+   src/speed_condition.cpp
 )
 
-add_dependencies(${PROJECT_NAME}
-  ${${PROJECT_NAME}_EXPORTED_TARGETS}
-  ${catkin_EXPORTED_TARGETS})
-
-target_link_libraries(${PROJECT_NAME}
-  ${catkin_LIBRARIES}
-  ${YAML_CPP_LIBRARIES})
-
-install(
-  TARGETS ${PROJECT_NAME}
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION})
-
-install(
-  DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+pluginlib_export_plugin_description_file(condition_plugins plugins.xml)
```

Comment:
is `plugins.xml` still installed?

Reply:
I think so, see the section "The ament index explained" in https://index.ros.org/doc/ros2/Tutorials/Ament-CMake-Documentation/. But I'm now wondering if the first argument is correct ... I don't really know how pluginlib works.

My impression is that these plugins are loaded by the `pluginlib::ClassLoader` in `scenario_conditions`, and maybe its [first constructor argument](http://docs.ros.org/en/noetic/api/pluginlib/html/classpluginlib_1_1ClassLoader.html#a65dcdf53dcc7be90f72d442423c08307) (the "package") identifies the folder that we install into here? That would mean that they need to be the same – what do you think @cvasfi @jilaada?

Reply:
Now it just remains to check that this plugin can indeed be loaded

Reply:
@mitsudome-r Do you know by chance whether the first argument needs to be `condition_plugins` or `scenario_conditions`?

Reply:
I did a little bit of digging myself - though I am not familiar with `pluginlib` either. I found this explanation: https://github.com/ros2/rosbag2/blob/master/docs/storage_plugin_development.md#writing-a-general-plugin
> The first argument rosbag2_storage denotes the library we add our plugin to (this will always be rosbag2_storage), while the second argument is the path to the plugin description file.

So I think what you have done here is correct

Reply:
I tried it out and it needs to be `scenario_conditions` indeed.